File: year2/systems-design-implementation/Lab4/Lab4/src/main/java/repo/intf/InscriereRepository.java
package repo.intf;

import domain.Inscriere;
import domain.Participant;
import domain.Proba;
import utils.Pair;

import java.util.List;

public interface InscriereRepository extends Repository<Pair<Integer, Integer>, Inscriere> {

    List<Participant> getInscrisi(Proba proba);

    List<Proba> getInscrieri(Participant participant);

    Integer nrParticipanti(Proba proba);
}

File: year2/systems-design-implementation/Lab4/Lab4Sharp/Repo/Interfaces/InscriereRepository.cs
using Lab2Sharp.Domain;
using Lab2Sharp.Utils;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab2Sharp.Repo.Interfaces
{
    public interface InscriereRepository : Repository<Pair<int, int>, Inscriere>
    {
        IEnumerable<Participant> GetInscrisi(Proba proba);
        IEnumerable<Proba> GetInscrieri(Participant participant);
        int NrParticipanti(Proba proba);
    }
}
